api/v1alpha1/database_webhook.go

```diff
@@ -188,10 +193,46 @@ func instanceSpecValidatorForCreate(instance *Instance, allErrs field.ErrorList,
 		))
 	}
 
+	// validating additional arguments
+	if additionalArgumentsErrors := instanceSpecAdditionalArgumentsValidator(instance, field.ErrorList{}, instancePath); additionalArgumentsErrors != nil {
+		allErrs = append(allErrs, additionalArgumentsErrors...)
+	}
+
 	databaselog.Info("Exiting instanceSpecValidatorForCreate...")
 	return allErrs
 }
 
+/* Checks if configured additional arguments are invalid */
+func isAdditionalArgumentsInvalid(typ string, additionalArguments map[string]string) (bool, map[string]bool) {
+	var allowedAdditionalArguments = util.GetAllowedAdditionalArgumentsForType(typ)
+
+	for name, _ := range additionalArguments {
+		if _, isPresent := allowedAdditionalArguments[name]; !isPresent {
+			return true, allowedAdditionalArguments
+		}
+	}
+
+	return false, allowedAdditionalArguments
+}
+
+/* Validates additional arguments */
+func instanceSpecAdditionalArgumentsValidator(instance *Instance, allErrs field.ErrorList, instancePath *field.Path) field.ErrorList {
```

Comment:
NIT: The name `instanceSpecAdditionalArgumentsValidator` sounds like an object/interface that would have a `validate` function. However, it is being used as a function. Can we change this to something like 
`ValidateAdditionalArguments`?

Reply:
Ok, makes sense.